libs/user-facing-errors/src/schema_engine.rs

```diff
@@ -286,6 +286,20 @@ pub struct ForeignKeyCreationNotAllowed;
 )]
 pub struct DirectDdlNotAllowed;
 
+#[derive(Debug, SimpleUserFacingError)]
+#[user_facing(
+    code = "P3023",
+    message = "When using an explicit schemas list in your datasource, `externalTables` in your prisma config must contain only fully qualified table names (e.g. `schema_name.table_name`)."
+)]
+pub struct MissingNamespaceInExternalTables;
+
+#[derive(Debug, SimpleUserFacingError)]
+#[user_facing(
+    code = "P3024",
+    message = "When using no explicit schemas list in your datasource, `externalTables` in your prisma config must contain only simple table names without a schema name."
+)]
+pub struct ExcessiveNamespaceInExternalTables;
```

Comment:
nitpick, but I'd probably call this `unexpected`
```suggestion
pub struct UnexpectedNamespaceInExternalTables;
```